libs/superagent/app/api/workflow_configs/saml_schema.py

```
@@ -1,17 +1,23 @@
-from typing import Any, Dict, List, Optional, Union
+from typing import Any, Optional, Union
 
 from pydantic import BaseModel
 
 
 class WorkflowDatasource(BaseModel):
     use_for: Optional[str]  # an alias for description
-    urls: Optional[List[str]]
+    urls: Optional[list[str]]
+
+
+class WorkflowSuperRag(WorkflowDatasource):
+    database_provider: Optional[str]  # for superrag
```

This should be set automatically by superagent. if you do not have a vector db connected this should throw an error.

---

yeah, but if a one have configured multiple databases, how we can know which one they want to choose?

---

Ok, we can leave it in.